paho/client.go

```diff
@@ -142,9 +169,20 @@ func NewClient(conf ClientConfig) *Client {
 	if c.PacketTimeout == 0 {
 		c.PacketTimeout = 10 * time.Second
 	}
-	if c.Router == nil {
-		c.Router = NewStandardRouter()
+
+	if c.Router == nil && len(c.onPublishReceived) == 0 {
+		c.Router = NewStandardRouter() // Maintain backwards compatibility (for now!)
```

Comment:
I'm wondering if it would be worth just avoiding the backward compatibility.

I understand that it's nice to avoid breaking users, but I'd also be wary of adding extra code paths for the sake of backward compatibility before this even comes out of beta. Beta is the chance to try things out and make changes when things don't work.

Reply:
@vishnureddy17 not really sure either way on this one. I think I'll push this change in and we can then look at removing the Router in a subsequent change (doing it this way means I don't need to update all of the examples etc in order to get this change in). We could remove the `Router` before the next release or, alternatively, wait for the following release (allowing users to update to the latest release and then remove references to the depreciated Router over a month or so).